posthog/models/hog_functions/hog_function.py

```diff
@@ -0,0 +1,115 @@
+import json
+from typing import Optional
+
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch.dispatcher import receiver
+
+from posthog.models.action.action import Action
+from posthog.models.team.team import Team
+from posthog.models.utils import UUIDModel
+from posthog.redis import get_client
+
+
+class HogFunction(UUIDModel):
+    team: models.ForeignKey = models.ForeignKey("Team", on_delete=models.CASCADE)
+    name: models.CharField = models.CharField(max_length=400, null=True, blank=True)
+    description: models.TextField = models.TextField(blank=True, default="")
+    created_at: models.DateTimeField = models.DateTimeField(auto_now_add=True, blank=True)
+    created_by: models.ForeignKey = models.ForeignKey("User", on_delete=models.SET_NULL, null=True, blank=True)
+    deleted: models.BooleanField = models.BooleanField(default=False)
+    updated_at: models.DateTimeField = models.DateTimeField(auto_now=True)
+    enabled: models.BooleanField = models.BooleanField(default=False)
+
+    hog: models.TextField = models.TextField()
+    bytecode: models.JSONField = models.JSONField(null=True, blank=True)
+
+    # TODO: Rename to "variables"
```

Comment:
I'm not sure this TODO is worth it... as the `input variables` can also be called `input`/`inputs`. Just calling `variables` loses some meaning, like going from "result" to "data" loses some meaning.